src/data/equiv/mul_add.lean

```diff
@@ -418,6 +418,23 @@ lemma Pi_congr_right_trans {η : Type*}
   (es : ∀ j, Ms j ≃* Ns j) (fs : ∀ j, Ns j ≃* Ps j) :
   (Pi_congr_right es).trans (Pi_congr_right fs) = (Pi_congr_right $ λ i, (es i).trans (fs i)) := rfl
 
+/-- A family indexed by a nonempty subsingleton type is equivalent to the element at the single
+index. -/
+@[to_additive add_equiv.Pi_subsingleton "A family indexed by a nonempty subsingleton type is
+equivalent to the element at the single index."]
+def Pi_subsingleton
+  {η : Type*} (M : η → Type*) [Π (j : η), mul_one_class (M j)] [subsingleton η] (i : η):
+  (Π (j : η), M j) ≃* M i :=
+{ map_mul' := λ f1 f2, pi.mul_apply _ _ _, ..equiv.Pi_subsingleton η M i }
```

Comment:
```suggestion
def Pi_subsingleton
  {ι : Type*} (M : ι → Type*) [Π j, mul_one_class (M j)] [subsingleton ι] (i : η) :
  (Π j, M j) ≃* M i :=
{ map_mul' := λ f1 f2, pi.mul_apply _ _ _, ..equiv.Pi_subsingleton M i }
```
(we usually use `ι` instead of `η`)

Reply:
I must have bad luck in my choice of cargo-culting-sources, because `η → Type*` is also used in a few places, but indeed far less often than `ι`. I guess I should adjust https://github.com/leanprover-community/mathlib/pull/11744 accordingly then? Ah, there I currently have `I`. But still.